client/src/components/files/FileUploadItem.vue

```diff
@@ -0,0 +1,271 @@
+      done: state === ImportState.FileImported,
+      cancelled: state === ImportState.Cancelled,
+      failed: state === ImportState.CreateFailed,
+    }"
+  >
+    <ion-item class="element ion-no-padding">
+      <div class="element-type">
+        <div class="element-type__icon">
+          <ms-image
+            v-if="file.icon"
+            :image="file.icon"
+            class="file-icon"
+          />
+        </div>
+      </div>
+      <div class="element-details">
+        <ion-text class="element-details__name body">
+          {{ file.name }}
+        </ion-text>
+        <ion-label class="element-details__size body-sm">
+          <span class="default-state">{{ file.size }}</span>
+          <span
+            class="hover-state"
+            v-show="[ImportState.FileImported].includes(state)"
+          >
+            {{ $t('FoldersPage.ImportFile.browse') }}
+          </span>
+        </ion-label>
+      </div>
+
+      <!-- done -->
+      <ion-icon
+        class="checkmark-icon default-state"
+        v-show="state === ImportState.FileImported"
+        :icon="checkmark"
+      />
+      <ion-icon
+        class="arrow-icon hover-state"
+        v-show="state === ImportState.FileImported"
+        :icon="arrowForward"
+      />
+
+      <!-- in progress -->
+      <ion-button
+        fill="clear"
+        size="small"
+        class="cancel-button"
+        v-if="state === ImportState.FileProgress"
+        @click="onCancelClick()"
+      >
+        <ion-icon
+          class="cancel-button__icon"
+          slot="icon-only"
+          :icon="close"
+        />
+      </ion-button>
+
+      <!-- cancel -->
+      <ion-text
+        class="cancel-text body"
+        v-if="state === ImportState.Cancelled"
+      >
+        {{ $t('FoldersPage.ImportFile.cancel') }}
+      </ion-text>
+
+      <!-- failed -->
+      <div
+        class="failed-content"
+        v-if="state === ImportState.CreateFailed"
+      >
+        <ion-text class="failed-text body">
+          {{ $t('FoldersPage.ImportFile.failed') }}
+        </ion-text>
+        <ms-information-tooltip
+          :text="$t('FoldersPage.ImportFile.failedDetails')"
+          class="information-icon"
+          slot="end"
+        />
+      </div>
+    </ion-item>
+    <ion-progress-bar
+      class="element-progress-bar"
+      :value="state === ImportState.Cancelled ? 100 : progress / 100"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { formatFileSize, getFileIcon, shortenFileName } from '@/common/file';
+import { MsImage } from '@/components/core/ms-image';
+import MsInformationTooltip from '@/components/core/ms-tooltip/MsInformationTooltip.vue';
+import { ImportData, ImportState } from '@/services/importManager';
+import { IonButton, IonIcon, IonItem, IonLabel, IonProgressBar, IonText } from '@ionic/vue';
+import { arrowForward, checkmark, close } from 'ionicons/icons';
+import { onMounted } from 'vue';
+
+let file = {
```

Comment:
Shouldn't it be a ref?

Something like
```typescript
const file: Ref<{icon: string, name: string, formattedSize: string} | null> = ref(null);

onMounted(() => {
  file.value = {icon: ...};
});
```

Also it's better to rename `size` to something like `formattedSize`. My first instinct was to say "Wait, a size should be a number, why is it a string??".